src/formatDate.js

```diff
@@ -50,7 +50,69 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  return toDate(toFormat, separateDate(fromFormat, date));
 }
 
 module.exports = formatDate;
+
+function separateDate(format, date) {
+  const dateSeparated = date.split(format[3]);
+
+  const dateGroup = {};
+
+  for (let i = 0; i < format.length; i++) {
+    if (format[i] === 'YYYY') {
+      dateGroup.year = dateSeparated[i];
+    }
+
+    if (format[i] === 'YY') {
+      dateGroup.year = dateSeparated[i];
+    }
+
+    if (format[i] === 'DD') {
+      dateGroup.day = dateSeparated[i];
+    }
+
+    if (format[i] === 'MM') {
+      dateGroup.month = dateSeparated[i];
+    }
+  }
+
+  return dateGroup;
+}
+
+function toDate(format, date) {
+  const finalDate = [];
+
+  for (let i = 0; i < format.length; i++) {
+    if (format[i] === 'YYYY') {
+      if (date.year.length === 4) {
+        finalDate.push(date.year);
+      }
```

Comment:
try to avoid nested `ifs`, they overcomplicate your code and make it more difficult to read